infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/VcsSshKeysProvisioner.java

```diff
@@ -171,17 +189,64 @@ private void mountSshKeySecret(String secretName, PodSpec podSpec) {
                         .build())
                 .build());
     List<Container> containers = podSpec.getContainers();
-    containers.forEach(
-        container -> {
-          VolumeMount volumeMount =
-              new VolumeMountBuilder()
-                  .withName(secretName)
-                  .withNewReadOnly(true)
-                  .withReadOnly(true)
-                  .withMountPath(SSH_PRIVATE_KEYS_PATH)
-                  .build();
-          container.getVolumeMounts().add(volumeMount);
-        });
+    VolumeMount sshKeysVolumeMount =
+        new VolumeMountBuilder()
+            .withName(secretName)
+            .withNewReadOnly(true)
+            .withReadOnly(true)
+            .withMountPath(SSH_PRIVATE_KEYS_PATH)
+            .build();
+    containers.forEach(container -> container.getVolumeMounts().add(sshKeysVolumeMount));
+
+    PodSecurityContext securityContext = podSpec.getSecurityContext();
+
+    /* The OpenSSH daemon requires 600 file mode access permissions for private keys.
+    Mounted keys have 640 permissions despite setting 'DefaultMode' to 600.
+    As a workaround copy the SSH keys and manually set them needed permissions.
+    The copy operation can be applied, if the primary group ID of the containers will be root (0).
+    TODO rework this when https://github.com/kubernetes/kubernetes/issues/81089 will be resolved */
+    if (securityContext != null
+        && securityContext.getFsGroup() == 0
+        && securityContext.getRunAsUser() == 0) {
+      copyPrivateKeysWithPermissions(podSpec, sshKeysVolumeMount);
+    }
+  }
+
+  // TODO remove this when https://github.com/kubernetes/kubernetes/issues/81089 will be resolved
+  private void copyPrivateKeysWithPermissions(PodSpec podSpec, VolumeMount sshKeysVolumeMount) {
+    String copiedSshKeysVolumeName = "copied-private-keys";
+    podSpec
+        .getVolumes()
+        .add(
+            new VolumeBuilder()
+                .withName(copiedSshKeysVolumeName)
+                .withNewEmptyDir()
+                .endEmptyDir()
+                .build());
+
+    VolumeMount copiedSshKeysVolumeMount =
+        new VolumeMountBuilder()
+            .withName(copiedSshKeysVolumeName)
+            .withMountPath(SSH_PRIVATE_KEYS_PATH + "-copy")
+            .build();
+
+    Container initContainer =
+        new ContainerBuilder()
+            .withName("ssh-private-keys-modifier")
+            .withImage("centos:centos7")
```

Comment:
This is bad practice and will be a nightmare for airgap users.

Reply:
Used a property